cmd/process-agent/collector.go

```diff
@@ -330,7 +330,12 @@ func (l *Collector) run(exit chan struct{}) error {
 		eventsEps = append(eventsEps, e.Endpoint.String())
 	}
 
-	var checkNames []string
+	checkNamesLength := len(l.enabledChecks)
+	if !ddconfig.Datadog.GetBool("process_config.disable_realtime_checks") {
+		checkNamesLength = checkNamesLength * 2
```

Comment:
Did you mean `+2` here?
```suggestion
		checkNamesLength = checkNamesLength + 2
```

Reply:
I made it twice the size because when realtime checks isn't disabled you add an extra name check on [line 344](https://github.com/DataDog/datadog-agent/blob/5f72c6b7c86d0b52100f7cb64aa643fdfd286655/cmd/process-agent/collector.go#L344). So worst case scenario, the total length will be twice the number of check names.

Reply:
We needed to add this workaround with `checkNamesLength` because `process` and `rt_process` are in fact running on the same `process` check. The only other check that supports realtime is `container` and in this case `rt_container` is a separate check that is already present in `enabledChecks`.

That being said, I think that it's ok to use `len(l.enabledChecks)*2` if eventually we add other realtime checks and consolidate `rt_container` into `container`.

What are you thoughts on this, @xornivore ?

Reply:
Agreed on `len(l.enabledChecks)*2` but let's add a comment.